crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -440,14 +448,15 @@ where
             // only one block to check and it's the current best block which we can fetch directly
             // Note: In case of a reorg, the best block's hash might have changed, hence we only
             // return early of we were able to fetch the best block's receipts
-            if let Some(receipts) = self.eth_cache.get_receipts(chain_info.best_hash).await? {
+            if let Some((block, receipts)) = self.eth_cache.get_block_and_receipts(chain_info.best_hash).await? {
```

Comment:
could you please add a note about
// perf: we're fetching the best block here which is expected to be cached